scripts/lib/CIME/case_st_archive.py

```diff
@@ -212,7 +212,7 @@ def _archive_restarts(case, archive, archive_entry,
     rundir = case.get_value("RUNDIR")
     casename = case.get_value("CASE")
     archive_restdir = join(dout_s_root, 'rest', datename)
-    if not datename_is_last or case.get_value('DOUT_S_SAVE_INTERIM_RESTART_FILES') \
+    if datename_is_last or case.get_value('DOUT_S_SAVE_INTERIM_RESTART_FILES') \
        or case.get_value("TEST"):
```

Comment:
Do we definitely need `case.get_value("TEST")` here? I'm wondering if that was added as a quick way to get the tests passing with the previous incorrect logic? It seems dangerous to have too many instances where logic differs depending on whether you're running a test, because it makes it more likely that we'll miss problems in testing. If this is needed, I'm fine keeping it, but let's remove it if possible.